dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50Coercion.war/src/io/openliberty/el50/fat/coercion/servlets/lambda/EL50LambdaExpressionCoercionServlet.java

```diff
@@ -68,30 +60,34 @@ public void testLambdaCoercion() throws Exception {
     }
 
     /**
-     * 
-     * Expression Language 5.0 in Jakarta EE10 Return the result of invoking the LambdaExpression with the parameters (coerced if necessary) that were passed to the Functional Interface method invocation.
+     *
+     * Expression Language 5.0 in Jakarta EE10 Return the result of invoking the LambdaExpression with the parameters (coerced if necessary) that were passed to the Functional
+     * Interface method invocation.
      *
      * Should fail since parameter of lambda was coerced to a wrong type then expected (int expected)
+     *
      * @throws Exception
      */
     @Test
     public void testLambdaCoercionWrongType() throws Exception {
         // Test lambda with a wrong type by passing a lambda evaluating with a wrong type
-        Object result = null;
-        boolean exceptionCaught = false;
-        try {
-            result = elp.eval("lambdaBean.testIntLambdaCoercion(param -> param.equals('should be int not string'))");
-        } catch (ELException e) {
-            exceptionCaught = true;
-        }
-        assertEquals("The expression did not throw expected ELException. Got result: "+result, true, exceptionCaught);
+        String result;
+        String expectedResult = "failed";
+
+        Object obj = elp.eval("lambdaBean.testIntLambdaCoercion(param -> param.equals('should be int not string'))");
+        assertNotNull(obj);
+        result = obj.toString();
+
+        assertEquals("The expression did not evaluate to: " + expectedResult + " but was: " + result, expectedResult, result);
```

Comment:
Linking to explanation of change: https://lists.apache.org/thread/cm0pycoyqq4pb1j816vdpdo4nds57dwj 9e (expand Mark's comment)